test/provision/compile.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
```

Comment:
If compilation fails, then currently ginkgo will silently ignore this and attempt to run the test anyway, then get into a loop like this:

```
level=info msg="Cilium status is false" test=RuntimeConnectivityTest
Error running command 'sudo cilium status': Process exited with status 1
```

Probably the individual test will time out after some time of doing this, but it'd be nice to fail out properly during vagrant provision if make or install fails.

Reply:
+1 to this. Please `set -e` at the top of this script and ensure that the provisioning of the vagrant scripts fails the build so that the tests fail-fast. I'm happy to make this a follow-up issue that we can distribute out after this is merged.